src/lib.rs

```diff
@@ -115,51 +115,56 @@ impl NonEmptyString {
     }
 }
 
-impl std::convert::AsRef<str> for NonEmptyString {
+impl AsRef<str> for NonEmptyString {
     fn as_ref(&self) -> &str {
         &self.0
     }
 }
 
-impl std::convert::AsRef<String> for NonEmptyString {
+impl AsRef<String> for NonEmptyString {
     fn as_ref(&self) -> &String {
         &self.0
     }
 }
 
-impl<'s> std::convert::TryFrom<&'s str> for NonEmptyString {
-    type Error = ();
+impl<'s> TryFrom<&'s str> for NonEmptyString {
+    type Error = &'s str;
 
     fn try_from(value: &'s str) -> Result<Self, Self::Error> {
         if value.is_empty() {
-            Err(())
-        } else {
-            Ok(NonEmptyString::new(value.to_owned()).expect("Value is not empty"))
+            return Err(value);
         }
+            
+        Ok(NonEmptyString(value.to_owned()))
     }
 }
 
-impl std::convert::TryFrom<String> for NonEmptyString {
+impl TryFrom<String> for NonEmptyString {
     type Error = String;
 
     fn try_from(value: String) -> Result<Self, Self::Error> {
         NonEmptyString::new(value)
     }
 }
 
+impl std::fmt::Display for NonEmptyString {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.0)
```

Comment:
I think it might be a bit nicer to just forward to the inner `String`
```suggestion
        std::fmt::Display::fmt(&self.0, f)
```